mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/MaildirFolder.java

```diff
@@ -53,11 +53,13 @@
 import org.apache.james.mailbox.model.MailboxACL.EntryKey;
 import org.apache.james.mailbox.model.MailboxACL.Rfc4314Rights;
 import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.UidValidity;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class MaildirFolder {
     private static final Logger LOGGER = LoggerFactory.getLogger(MaildirFolder.class);
+    private static final UidValidity DEFAULT_UID_VALIDITY = UidValidity.of(-1);
```

Comment:
this is invalid, you probably needs a "Null object" (from Null Object Pattern)

Reply:
Why not an optional?

Reply:
Optional is the poor-man-null-object. It may be enough, yes.

Reply:
I disagree having been fighting null object that implicitly require some specific handling.
With oprional it is explicit

Reply:
> I disagree having been fighting null object that implicitly require some specific handling.
With oprional it is explicit

You probably don't use null object pattern as it should because Optional uses null object pattern for a reference, there's nothing special in Optional that you would get from any other null object.